Build failure: sratoolkit

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Build of sratoolkit package on :
```
    "nixpkgs": {
      "locked": {
        "lastModified": 1751274312,
        "narHash": "sha256-/bVBlRpECLVzjV19t5KMdMFWSwKLtb5RyXdjz3LJT+g=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "50ab793786d9de88ee30ec4e4c24fb4236fc2674",
        "type": "github"
      }
    }
```
done via:
```
nix-shell -p sratoolkit
```
ends in error:

```
make: *** [Makefile:166: all] Error 2
error: builder for '/nix/store/m8svyydlxj9q8jhdgdhrd6b0b5ci1nm0-sratoolkit-3.2.1.drv' failed with exit code 2;
       last 25 log lines:
       > 115649 |   return pNew;
       >        |          ^~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:115609:10: note: declared here
       > 115609 |   Select standin;
       >        |          ^~~~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:115609:10: note: declared here
       > /build/source/libs/vdb-sqlite/sqlite3.c: In function 'sqlite3VdbeExpandSql':
       > /build/source/libs/vdb-sqlite/sqlite3.c:77729:10: warning: function may return address of local variable [-Wreturn-local-addr]
       > 77729 |   return sqlite3StrAccumFinish(&out);
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > 77625 |   char zBase[100];         /* Initial working space */
       >       |        ^~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77729:10: warning: function may return address of local variable [-Wreturn-local-addr]
       > 77729 |   return sqlite3StrAccumFinish(&out);
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > 77625 |   char zBase[100];         /* Initial working space */
       >       |        ^~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > [ 39%] Linking C static library ../../ilib/libvdb-sqlite.a
       > [ 39%] Built target vdb-sqlite
       > make: *** [Makefile:166: all] Error 2
       For full logs, run:
         nix log /nix/store/m8svyydlxj9q8jhdgdhrd6b0b5ci1nm0-sratoolkit-3.2.1.drv
```

Before this final error of the build, another one appears during the build:

```
/build/source/libs/kxml/xml.c:174:52: error: expected ';', ',' or ')' before 'ATTRIBUTE_UNUSED'
  174 | static void s_xmlGenericErrorDefaultFunc(void *ctx ATTRIBUTE_UNUSED,
      |                                                    ^~~~~~~~~~~~~~~~
/build/source/libs/kxml/xml.c: In function 'KXMLMgrMakeDocReadFromMemory':
/build/source/libs/kxml/xml.c:266:35: error: 's_xmlGenericErrorDefaultFunc' undeclared (first use in this function); did you mean 'initGenericErrorDefaultFunc'?
  266 |     {   xmlSetGenericErrorFunc(0, s_xmlGenericErrorDefaultFunc); }
      |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                   initGenericErrorDefaultFunc
/build/source/libs/kxml/xml.c:266:35: note: each undeclared identifier is reported only once for each function it appears in
```


### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
/build/source/libs/kxml/xml.c:174:52: error: expected ';', ',' or ')' before 'ATTRIBUTE_UNUSED'
  174 | static void s_xmlGenericErrorDefaultFunc(void *ctx ATTRIBUTE_UNUSED,
      |                                                    ^~~~~~~~~~~~~~~~
/build/source/libs/kxml/xml.c: In function 'KXMLMgrMakeDocReadFromMemory':
/build/source/libs/kxml/xml.c:266:35: error: 's_xmlGenericErrorDefaultFunc' undeclared (first use in this function); did you mean 'initGenericErrorDefaultFunc'?
  266 |     {   xmlSetGenericErrorFunc(0, s_xmlGenericErrorDefaultFunc); }
      |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                   initGenericErrorDefaultFunc
/build/source/libs/kxml/xml.c:266:35: note: each undeclared identifier is reported only once for each function it appears in
...
make: *** [Makefile:166: all] Error 2
error: builder for '/nix/store/m8svyydlxj9q8jhdgdhrd6b0b5ci1nm0-sratoolkit-3.2.1.drv' failed with exit code 2;
       last 25 log lines:
       > 115649 |   return pNew;
       >        |          ^~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:115609:10: note: declared here
       > 115609 |   Select standin;
       >        |          ^~~~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:115609:10: note: declared here
       > /build/source/libs/vdb-sqlite/sqlite3.c: In function 'sqlite3VdbeExpandSql':
       > /build/source/libs/vdb-sqlite/sqlite3.c:77729:10: warning: function may return address of local variable [-Wreturn-local-addr]
       > 77729 |   return sqlite3StrAccumFinish(&out);
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > 77625 |   char zBase[100];         /* Initial working space */
       >       |        ^~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77729:10: warning: function may return address of local variable [-Wreturn-local-addr]
       > 77729 |   return sqlite3StrAccumFinish(&out);
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > 77625 |   char zBase[100];         /* Initial working space */
       >       |        ^~~~~
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > /build/source/libs/vdb-sqlite/sqlite3.c:77625:8: note: declared here
       > [ 39%] Linking C static library ../../ilib/libvdb-sqlite.a
       > [ 39%] Built target vdb-sqlite
       > make: *** [Makefile:166: all] Error 2
       For full logs, run:
         nix log /nix/store/m8svyydlxj9q8jhdgdhrd6b0b5ci1nm0-sratoolkit-3.2.1.drv
```

### Additional context

Before submitting this issue, I opened a [discussion](https://discourse.nixos.org/t/possible-issue-sratoolkit-package-build-failure/67438) on [nixos discourse](https://discourse.nixos.org/). Other users were able to reproduce this.

# Last working nixpkgs version
```
    "nixpkgs": {
      "locked": {
        "lastModified": 1746422338,
        "narHash": "sha256-NTtKOTLQv6dPfRe00OGSywg37A1FYqldS6xiNmqBUYc=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "5b35d248e9206c1f3baf8de6a7683fee126364aa",
        "type": "github"
      }
    },
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.40, NixOS, 25.11 (Xantusia), 25.11.20250728.dc96378`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/fy1zkcaf993w05ylfqx93b27am49i27b-source`

### Notify maintainers

@thyol @t4ccer

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sr/sratoolkit/attribute_unused.patch
+++ b/pkgs/by-name/sr/sratoolkit/attribute_unused.patch
@@ -1,0 +1,14 @@
+diff --git a/libs/kxml/xml.c b/libs/kxml/xml.c
+index ce445424..41e21612 100644
+--- a/libs/kxml/xml.c
++++ b/libs/kxml/xml.c
+@@ -46,6 +46,9 @@ struct s_KNodeNamelist;
+ #include <assert.h>
+ #include <string.h>
+ 
++#ifndef ATTRIBUTE_UNUSED
++#define ATTRIBUTE_UNUSED
++#endif
+ 
+ #define XML_DEBUG(msg) DBGMSG (DBG_XML, DBG_FLAG(DBG_XML_XML), msg)
+ 

--- a/pkgs/by-name/sr/sratoolkit/package.nix
+++ b/pkgs/by-name/sr/sratoolkit/package.nix
@@ -25,6 +25,8 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-OeM4syv9c1rZn2ferrhXyKJu68ywVYwnHoqnviWBZy4=";
   };
 
+  patches = [ ./attribute_unused.patch ];
+
   cmakeFlags = [
     "-DVDB_INCDIR=${ncbi-vdb}/include"
     "-DVDB_LIBDIR=${ncbi-vdb}/lib"

